binary-codec/types/amount_test.go

```diff
@@ -155,6 +155,65 @@ func TestVerifyIOUValue(t *testing.T) {
 	}
 }
 
+func TestSerializeXrpAmount(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		input          string
+		expectedOutput []byte
+		expErr         error
+	}{
+		{
+			name:           "valid xrp value - 1",
+			input:          "524801",
+			expectedOutput: []byte{0x40, 0x00, 0x00, 0x00, 0x00, 0x8, 0x2, 0x01},
+			expErr:         nil,
+		},
+		{
+			name:           "valid xrp value - 2",
+			input:          "7696581656832",
+			expectedOutput: []byte{0x40, 0x00, 0x7, 0x00, 0x00, 0x4, 0x1, 0x00},
+			expErr:         nil,
+		},
+		{
+			name:           "valid xrp value - 3",
+			input:          "10000000",
+			expectedOutput: []byte{0x40, 0x00, 0x00, 0x00, 0x00, 0x98, 0x96, 0x80},
+			expErr:         nil,
+		},
+		{
+			name:           "invalid xrp value - negative",
+			input:          "-125000708",
+			expectedOutput: nil,
+			expErr:         errors.New("XRP value is an invalid XRP amount"),
+		},
+		{
+			name:           "invalid xrp value - decimal",
+			input:          "125000708.0",
+			expectedOutput: nil,
+			expErr:         errors.New("XRP value must not contain a decimal"),
+		},
+		{
+			name:           "invalid xrp value - out of range",
+			input:          "983496373783750346983046704237509273095720937560927469729076023947673406",
+			expectedOutput: nil,
+			expErr:         errors.New("XRP value is an invalid XRP amount"),
```

Comment:
Can we add some boundary tests? One just before the boundary, on the boundary(max value) and over the boundary?

Reply:
done thanks